README.md

```diff
@@ -1064,545 +983,270 @@ helpLinks: {
-
-    Valid OIDC scopes: `openid`, `email`, `profile`, `address`, `phone`
-
-    ```javascript
-    authParams: {
-      scopes: ['openid', 'email', 'profile', 'address', 'phone']
-    }
-    ```
-
-- **authParams.state:** Specify a state that will be validated in an [OAuth][] response. This is usually only provided during redirect flows to obtain an authorization code. Defaults to a random string. This value can be retrieved from the URL on the login redirect [callback][]. For more information on handling the redirect [callback][], see [authClient.token.parseFromUrl()](https://github.com/okta/okta-auth-js#tokenparsefromurloptions)
+```javascript
+postSubmit: (response, onSuccess) => {
+  // This example will log the API request body to the browser console before completing registration.
+  console.log(response);
+  onSuccess(response);
+}
+```
 
-    ```javascript
-    authParams: {
-      state: '8rFzn3MH5q'
-    }
-    ```
+#### Handling registration callback errors
 
-- **authParams.nonce:** Specify a nonce that will be validated in an id_token. This is usually only provided during redirect flows to obtain an authorization code that will be exchanged for an id_token. Defaults to a random string.
+- **onFailure and ErrorObject:** The onFailure callback accepts an error object that can be used to show a form level vs field level error on the registration form.
 
-    ```javascript
-    authParams: {
-      nonce: '51GePTswrm'
-    }
-    ```
+##### Use the default error
 
-- **authParams.issuer:** Specify a custom issuer to perform the [OIDC][] flow. Defaults to the baseUrl plus "/oauth2/default". See the guide on setting up a [Custom Authorization Server][] for more information.
+```javascript
+preSubmit: (postData, onSuccess, onFailure) => {
+  // A generic form level error is shown if no error object is provided
+  onFailure();
+}
+```
 
-    ```javascript
-    authParams: {
-      issuer: 'https://{yourOktaDomain}/oauth2/default'
-    }
-    ```
+##### Display a form error
 
-- **authParams.authorizeUrl:** Specify a custom authorizeUrl to perform the [OIDC][] flow. Defaults to the issuer plus "/v1/authorize".
+  ```javascript
+preSubmit: (postData, onSuccess, onFailure) => {
+  const error = {
+    "errorSummary": "Custom form level error"
+  };
+  onFailure(error);
+}
+```
 
-    ```javascript
-    authParams: {
-      issuer: 'https://{yourOktaDomain}/oauth2/default',
-      authorizeUrl: 'https://{yourOktaDomain}/oauth2/default/v1/authorize'
-    }
-    ```
+##### Display a form field error
 
-- **authScheme:** Authentication scheme for [OIDC][] [authentication][]. You will normally not need to override this value. Defaults to `OAUTH2`.
+```javascript
+  preSubmit: (postData, onSuccess, onFailure) => {
+    const error = {
+        "errorSummary": "API Error",
+        "errorCauses": [
+            {
+              "errorSummary": "Custom field level error",
+              "property": "userProfile.email",
```

Comment:
is this how it works ? did we start using `property` field ? I doubt IDX APIs return errors with `property` field

Reply:
Good catch, I'm not entirely sure about this section, I copied it from our `interaction_code_flow` document, but will have to confirm its accuracy. @sherizada-okta do you know if this is accurate?

Reply:
This is an example of a custom error that the user can create if they want to show their own error. This is the format which courage can translate.